### PR TITLE
Feature/cse15 tx summary

### PR DESCRIPTION
**Pull request by flyingleafe**
`ifdef`-carnival aimed to store and maintain additional info about transactions (there's basically only source inputs and place in block yet).

### CODE REVIEW

File: src/Pos/Types/Explorer.hs [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+       ( TxExtra (..)
+       ) where
+
+import           Universum
+
+import           Pos.Types.Core (Address, HeaderHash)
+
+data TxExtra = TxExtra
+    { teBlockchainPlace :: Maybe (HeaderHash, Word32)
+    , teInputAddresses  :: [Address]
```
Comment (pva701) [inline src/Pos/Types/Explorer.hs]: is it really needed list here? maybe `NonEmpty`?

File: src/Pos/Txp/Logic/Local.hs [hunk 2]
```diff
@@ -93,7 +94,12 @@ txNormalize = do
     res <- runExceptT $
            runDBTxp $
            execTxpTLocal def def def $
-           normalizeTxp $ HM.toList _mpLocalTxs
+           normalizeTxp $
+#ifdef DWITH_EXPLORER
+           HM.toList $ HM.intersectionWith (,) _mpLocalTxs _mpLocalTxsExtra
```
Comment (pva701) [inline src/Pos/Txp/Logic/Local.hs]: why is it intersection needed here? can you explain pls

File: src/Pos/Txp/Toil/Logic.hs [hunk 3]
```diff
@@ -32,6 +33,9 @@ import           Pos.Txp.Toil.Class   (MonadBalances (..), MonadBalancesRead (..
                                        MonadTxPool (..), MonadUtxo (..))
 import           Pos.Txp.Toil.Failure (TxpVerFailure (..))
 import qualified Pos.Txp.Toil.Utxo    as Utxo
+#ifdef DWITH_EXPLORER
+import Pos.Types.Explorer (TxExtra)
```
Comment (pva701) [inline src/Pos/Txp/Toil/Logic.hs]: strange aligning, gromak will burn

File: src/Pos/Txp/Logic/Local.hs (hunk 2 above)
Comment (pva701) [inline src/Pos/Txp/Logic/Local.hs]: nvm

File: src/Pos/Txp/Toil/Trans.hs [hunk 4]
```diff
@@ -101,7 +102,7 @@ instance MonadBalancesRead m => MonadBalances (TxpT m) where
 instance Monad m => MonadTxPool (TxpT m) where
     hasTx id = TxpT $ use $ txmMemPool . mpLocalTxs . to (HM.member id)
 
-#ifdef DWITH_EXPLORER
+#ifdef WITH_EXPLORER
     putTxWithUndo id tx undo extra = TxpT $ do
```
Comment (pva701) [inline src/Pos/Txp/Toil/Trans.hs]: why `MonadTxExtra` exists and this method takes extra at the same time?
seems after creating of `MonadTxExtra`, extra parameter isn't needed more.

File: src/Pos/Block/Logic/Internal.hs [hunk 5]
```diff
@@ -1,3 +1,4 @@
+{-# LANGUAGE CPP                 #-}
```
Comment (gromakovsky) [inline src/Pos/Block/Logic/Internal.hs]: Is it needed? Redundant CPP is bad.

File: src/Pos/Txp/Logic/Local.hs [hunk 6]
```diff
@@ -56,21 +60,30 @@ txProcessTransaction itw@(txId, (UnsafeTx{..}, _, _)) = do
         Left er -> do
             logDebug $ sformat ("Transaction processing failed: "%build) txId
             throwError er
-        Right _   ->
+        Right _   -> do
```
Comment (gromakovsky) [inline src/Pos/Txp/Logic/Local.hs]: Redundant, but I don't really care.

File: src/Pos/Types/Explorer.hs [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+       ) where
+
+import           Universum
+
+import           Pos.Core.Types     (HeaderHash)
+import           Pos.Txp.Core.Types (TxOut)
+
+data TxExtra = TxExtra
+    { teBlockchainPlace :: Maybe (HeaderHash, Word32)
+    , teInputOutputs    :: [TxOut]
```
Comment (gromakovsky) [inline src/Pos/Types/Explorer.hs]: I guess `TxOutAux` may be needed sooner or later.

File: src/Pos/Block/Logic/Internal.hs (hunk 5 above)
Comment (flyingleafe) [inline src/Pos/Block/Logic/Internal.hs]: Now - yes

**Comment by gromakovsky** [comment]
I don't know what meaning timestamp should have but I assume you know what you do.

**Pull request merged**